crates/red_knot_python_semantic/src/types/mro.rs

```diff
@@ -377,6 +377,7 @@ impl<'db> ClassBase<'db> {
                 | KnownInstanceType::Union
                 | KnownInstanceType::NoReturn
                 | KnownInstanceType::Never
+                | KnownInstanceType::Any
```

Comment:
It's under-specified, and perhaps unfortunate, but the typing spec does actually permit inheriting from `Any`: https://typing.readthedocs.io/en/latest/spec/special-types.html#any

And this is used in typeshed (e.g. for `unittest.mock.Mock`): https://github.com/python/typeshed/blob/main/stdlib/unittest/mock.pyi#L109

This is already supported by the `ClassBase` enum: we should return `Some(Self::Any)` instead of `None` for `KnownInstanceType::Any`.